src/ode_solver/low_storage_runge_kutta_ode_solver.cpp

```diff
@@ -0,0 +1,270 @@
+        // Apply limiter at every RK stage
+        if (this->limiter) {
+            this->limiter->limit(this->dg->solution,
+                this->dg->dof_handler,
+                this->dg->fe_collection,
+                this->dg->volume_quadrature_collection,
+                this->dg->high_order_grid->fe_system.tensor_degree(),
+                this->dg->max_degree,
+                this->dg->oneD_fe_collection_1state,
+                this->dg->oneD_quadrature_collection);
+        }
+        this->dg->assemble_residual();
+        this->dg->apply_inverse_global_mass_matrix(this->dg->right_hand_side, rhs);
+        rhs *= dt;
+        storage_register_4.add(this->butcher_tableau->get_b_hat(n_rk_stages), rhs);       
+    }
+    this->dg->solution = storage_register_1;
+
+    // Apply limiter at every RK stage
+    if (this->limiter) {
+        this->limiter->limit(this->dg->solution,
+            this->dg->dof_handler,
+            this->dg->fe_collection,
+            this->dg->volume_quadrature_collection,
+            this->dg->high_order_grid->fe_system.tensor_degree(),
+            this->dg->max_degree,
+            this->dg->oneD_fe_collection_1state,
+            this->dg->oneD_quadrature_collection);
+        }
+
+    if ((this->ode_param.ode_output) == Parameters::OutputEnum::verbose &&
+        (this->current_iteration%this->ode_param.print_iteration_modulo) == 0 ) {
+        this->pcout << " Evaluating system update... " << std::endl;
+    }
+ 
+    this->pcout << std::endl;
+    ++(this->current_iteration);
+    this->current_time += dt;
+    this->pcout << " Time is: " << this->current_time <<std::endl;
+    this->pcout << std::endl;
+}
+
+template <int dim, typename real, int n_rk_stages, typename MeshType> 
+double LowStorageRungeKuttaODESolver<dim,real,n_rk_stages, MeshType>::get_automatic_error_adaptive_step_size (real dt, const bool /*pseudotime*/)
+{
+    double error = 0.0;
+    w = 0.0;
+
+    // error based step size 
+    if (is_3Sstarplus == false){
+        // loop sums elements at each mpi processor
+        for (dealii::LinearAlgebra::distributed::Vector<double>::size_type i = 0; i < storage_register_1.local_size(); ++i) {
+            error = storage_register_1.local_element(i) - storage_register_2.local_element(i);
+            w = w + pow(error / (atol + rtol * std::max(std::abs(storage_register_1.local_element(i)), std::abs(storage_register_2.local_element(i)))), 2);
+        }
+    }
+    else if (is_3Sstarplus == true){
+        // loop sums elements at each mpi processor
+        for (dealii::LinearAlgebra::distributed::Vector<double>::size_type i = 0; i < storage_register_1.local_size(); ++i) {
+            error = storage_register_1.local_element(i) - storage_register_4.local_element(i);
+            w = w + pow(error / (atol + rtol * std::max(std::abs(storage_register_1.local_element(i)), std::abs(storage_register_4.local_element(i)))), 2);
+        }
+    }
+    this->pcout << std::endl;
```

Comment:
Would this std::endl statement result in unnecessary new lines since there is no other print statements within this function?